parlai/crowdsourcing/tasks/pairwise_per_turn_eval/per_turn_eval_blueprint.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import yaml
+from dataclasses import dataclass, field
+from typing import Dict, TYPE_CHECKING
+
+from mephisto.abstractions.blueprint import SharedTaskState
+from mephisto.operations.registry import register_mephisto_abstraction
+from omegaconf import DictConfig
+
+from parlai.crowdsourcing.tasks.model_chat.model_chat_blueprint import (
+    ModelChatBlueprint,
+    ModelChatBlueprintArgs,
+    BaseModelChatBlueprint,
+)
+from parlai.crowdsourcing.tasks.pairwise_per_turn_eval.bot_agent import (
+    PerTurnEvalTurkLikeAgent,
+)
+
+if TYPE_CHECKING:
+    from mephisto.data_model.task import TaskRun
+
+
+def get_task_path():
+    return os.path.dirname(os.path.realpath(__file__))
+
+
+BLUEPRINT_TYPE = 'per_turn_eval_blueprint'
+
+
+@dataclass
+class PerTurnEvalBlueprintArgs(ModelChatBlueprintArgs):
+    _blueprint_type: str = BLUEPRINT_TYPE
+
+    custom_source_dir: str = field(
+        default=os.path.join(get_task_path(), 'frontend'),
+        metadata={"help": "Path to frontend code"},
+    )
+
+    random_seed: int = field(
+        default=42, metadata={"help": 'Seed for random operations'}
+    )
+
+    task_question: str = field(
+        default="Which next response from your partner would be more interesting?",
+        metadata={"help": "The task question to choose bot's responses based on"},
+    )
+
+    task_config_path: str = field(
+        default=os.path.join(get_task_path(), 'task_config'),
+        metadata={"help": "Base path to pull task configuration information"},
+    )
+
+    task_description_file: str = field(
+        default="${mephisto.blueprint.task_config_path}/task_description.html",
+        metadata={"help": "Path to file of HTML to show on the task-description page"},
+    )
+
+    left_pane_text_path: str = field(
+        default="${mephisto.blueprint.task_config_path}/left_pane_text.html",
+        metadata={
+            "help": "Path to file of HTML to show on the left-hand pane of the chat window"
+        },
+    )
+
+
+@register_mephisto_abstraction()
+class PerTurnEvalBlueprint(ModelChatBlueprint):
+    """
+    Blueprint for per turn evaluation method.
+
+    This blueprint subclasses ModelChatBlueprint to provide logic for keeping track of
+    how many more conversations are needed per model; this logic is not shared with
+    other blueprints.
+    """
+
+    ArgsClass = PerTurnEvalBlueprintArgs
+    # SharedStateClass = SharedModelChatTaskState
```

Comment:
I'd remove this commented-out line too

Reply:
Sure, though there will be a follow up PR that cleans up each file - this PR mainly looks at high level changes